syntax = "proto3";
package ledger.core.message;

enum ServiceRequestType {
    HTTP_REQ = 0;
}

message ServiceRequest {
     ServiceRequestType type = 1;
     bytes request_body = 2;
}

message ServiceResponse {
     string error = 1;
     bytes response_body = 2;
}

message HttpRequest {
    string method = 1;
    string url = 2;
    map<string, string> headers = 3;
    string body = 4;
}

message HttpResponse {
    int32 code = 1;
    string body = 2;
}
